pygls/lsp/types/workspace.py

```diff
@@ -0,0 +1,200 @@
+
+
+class WorkspaceFolder(Model):
+    uri: str
+    name: str
+
+
+class WorkspaceFoldersChangeEvent(Model):
+    added: List[WorkspaceFolder]
+    removed: List[WorkspaceFolder]
+
+
+class DidChangeWorkspaceFoldersParams(Model):
+    event: WorkspaceFoldersChangeEvent
+
+
+class DidChangeConfigurationClientCapabilities(Model):
+    dynamic_registration: Optional[bool] = False
+
+
+class DidChangeConfigurationParams(Model):
+    settings: Any
+
+
+class ConfigurationItem(Model):
+    scope_uri: Optional[str] = None
+    section: Optional[str] = None
+
+
+class ConfigurationParams(Model):
+    items: List[ConfigurationItem]
+
+
+class DidChangeWatchedFilesClientCapabilities(Model):
+    dynamic_registration: Optional[bool] = False
+
+
+class WatchKind(enum.IntFlag):
+    Create = 1
+    Change = 2
+    Delete = 4
+
+
+class FileSystemWatcher(Model):
+    glob_pattern: str
+    kind: Optional[WatchKind] = WatchKind.Create | WatchKind.Change | WatchKind.Delete
+
+
+class DidChangeWatchedFilesRegistrationOptions(Model):
+    watchers: List[FileSystemWatcher]
+
+
+class FileChangeType(enum.IntEnum):
+    Created = 1
+    Changed = 2
+    Deleted = 3
+
+
+class FileEvent(Model):
+    uri: str
+    type: FileChangeType
+
+
+class DidChangeWatchedFilesParams(Model):
+    changes: List[FileEvent]
+
+
+class WorkspaceSymbolClientCapabilities(Model):
+    dynamic_registration: Optional[bool] = False
+    symbol_kind: Optional[WorkspaceCapabilitiesSymbolKind] = None
+
+
+class WorkspaceSymbolOptions(WorkDoneProgressOptions):
+    pass
+
+
+class WorkspaceSymbolRegistrationOptions(WorkspaceSymbolOptions):
+    pass
+
+
+class WorkspaceSymbolParams(WorkDoneProgressParams, PartialResultParams):
+    query: str
+
+
+class ExecuteCommandClientCapabilities(Model):
+    dynamic_registration: Optional[bool] = False
+
+
+class ExecuteCommandOptions(WorkDoneProgressOptions):
+    commands: List[str]
+
+
+class ExecuteCommandRegistrationOptions(ExecuteCommandOptions):
+    pass
+
+
+class ExecuteCommandParams(WorkDoneProgressParams):
+    command: str
+    arguments: Optional[List[Any]] = None
+
+
+class ApplyWorkspaceEditParams(Model):
+    edit: WorkspaceEdit
+    label: Optional[str] = None
+
+
+class ApplyWorkspaceEditResponse(Model):
+    applied: bool
+    failure_reason: Optional[str] = None
+
+
+class DidOpenTextDocumentParams(Model):
+    text_document: TextDocumentItem
+
+
+class TextDocumentContentChangeEvent(Model):
+    range: Optional[Range] = None
+    range_length: Optional[NumType] = None
+    text: str = ''
+
+
+class TextDocumentContentChangeTextEvent(Model):
+    text: str
+
+
+class DidChangeTextDocumentParams(Model):
+    text_document: VersionedTextDocumentIdentifier
+    content_changes: Union[List[TextDocumentContentChangeEvent],
+                           List[TextDocumentContentChangeTextEvent]]
+
+
+class WillSaveTextDocumentParams(Model):
+    text_document: TextDocumentIdentifier
+    reason: int
```

Comment:
Is this `reason` `int` ? according to https://microsoft.github.io/language-server-protocol/specification#textDocument_willSave it should be `TextDocumentSaveReason`